rclcpp/wait_set/wait_set_random_order.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021, Apex.AI Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+
+#include "rclcpp/rclcpp.hpp"
+#include "std_msgs/msg/string.hpp"
+
+#include "random_listener.hpp"
+#include "random_talker.hpp"
+
+using namespace std::chrono_literals;
+
+/* For this example, we will be creating a talker node with three publishers which will
+ * publish the topics A, B, C in random order each time. The order in which the messages are
+ * handled is defined deterministically by the user in the code using a wait-set based loop.
+ * That is, in this example we always take and process the data in the same order  A, B, C
+ * regardless of the arrival order. */
+
+int32_t main(const int32_t argc, char ** const argv)
+{
+  rclcpp::init(argc, argv);
+
+  auto listener = std::make_shared<Listener>();
```

Comment:
I think you still need to spin the listener, even if you're waiting on the subscriptions with a wait set.

Reply:
Is this required as a best practice, like for instance to be consistent with the other examples or is there an specific reason to spin the node in this case?

Reply:
I think it is a best practice, but for example, if you're using it with `use_sim_time = true` then you need to spin it to have time move forward.